src/routes/blog/index.svelte

```diff
@@ -1,9 +1,25 @@
 <script lang="ts" context="module">
   import type { LoadInput, LoadOutput } from '@sveltejs/kit/types/internal';
+  import TagField from '$lib/components/utils/acm-tagfield.svelte';
 
-  export async function load({ fetch }: LoadInput): Promise<LoadOutput> {
-    const response = await fetch(`/blog.json`);
-    return { props: { posts: await response.json() } };
+  let tags = [];
+  let selectedTags: string[] = [];
+
+  export async function load(event: LoadInput): Promise<LoadOutput> {
+    const target = new URL('/blog.json', event.url);
+
+    if (event.url.searchParams.has('l') && event.url.searchParams.get('l').length > 0) {
+      target.searchParams.set('l', event.url.searchParams.get('l'));
+      selectedTags = event.url.searchParams.get('l').split(',');
+    } else {
+      selectedTags = [];
+    }
+
+    const response = await fetch(target.toString());
+    const blogOutput = await response.json();
+    tags = blogOutput.tags;
+
+    return { props: { posts: blogOutput.posts } };
```

Comment:
We must pass information to the client by returning it as a returned property of the `load` function.
```suggestion
    const target = new URL('/blog.json', url);
    const rawLabels = url.searchParams.has('l');
    
    const selectedTags = [];
    if (rawLabels !== null && rawLabels.length > 0) {
      target.searchParams.set('l', rawLabels);
      selectedTags.push(...rawLabels.split(','));
    }

    const response = await fetch(target.toString());
    const { posts, tags } = await response.json();

    return { props: { posts, tags, selectedTags } };
```

Reply:
I fixed it.